### ISSUE TITLE
Hooks and commands withs spaces

### ISSUE THREAD

**Issue by coderforlife** [TheLocehiliosan/yadm#187]
The variable `$YADM_HOOK_FULL_COMMAND` loses information about spaces. For example, if the original command was

```
yadm add "Hello World"
```

The hook will see 4 values in that string (without the quotes): `yadm`, `add`, `Hello`, and `World`.

**Comment by TheLocehiliosan** [comment]
Good catch. I think this should be pretty easy to fix. Something like this will do it.

```
# capture full command, for passing to hooks
for param in "$@"; do
  FULL_COMMAND="${FULL_COMMAND} $(printf '%q' "$param")"
done
FULL_COMMAND="${FULL_COMMAND# }"
```

Needs to get cleaned up, and tests written for it, but should be fairly simple.

**Comment by TheLocehiliosan** [comment]
This change has been published to the `develop` branch and will be included in the next release.

I tried several different approaches to this issue, the main problem being there is not any standard way to represent an array in an environment variable. In the end, I've decided to keep parameters in a single exported variable, continue to space delimit the parameters, but escape whitespace with a backslash. Any hooks will need to distinguish between escaped, and unescaped whitespace.
